### PR TITLE
Added orientation acceleration to message BaseMoving (solves #279)

### PR DESCRIPTION
**Pull request by TBreitenbergerConti**

### CODE REVIEW

File: osi_common.proto [hunk 1]
```diff
@@ -395,4 +395,19 @@ message BaseMoving
...
     repeated Vector2d base_polygon = 7;
+    
+    // The relative orientation acceleration of the moving object w.r.t. its parent
+    // frame and parent orientation acceleration in the center point of the bounding box
+    // (origin of the bounding box frame), noted in the parent frame.
+    // The orientation becomes global/absolute if the parent frame is inertial
+    // (all parent frames up to ground truth).
+    //
+    // <tt>Rotation_yaw_pitch_roll(#orientation (t)) :=
+    // Rotation_yaw_pitch_roll(#orientation_rate
```
Comment (LudwigFriedmannBMW) [inline osi_common.proto]: Please provide an example for orientation_acceleration here.
Moreover, the last multiply should be a plus?
That seems to be wrong in the documentation of orientation rate...

**Comment by LudwigFriedmannBMW** [comment]
Hi @TBreitenbergerConti and @CarloVanDriestenBMW,

as matrix multiplication on every timestep results in unnecessary numerical drift and as we're neither interested in relative kinematics nor in time derivatives of rotational matrices - why don't we use examples for the scalar values (which I would recommmend to update)?

For discrete time steps and no higher order integration schemes, the linear scalar formulas would be:

orientation.yaw(t) = orientation_rate.yaw(t) * dt + orientation.yaw(t-dt)
orientation.pitch(t) = orientation_rate.pitch(t) * dt + orientation.pitch(t-dt)
orientation.roll(t) = orientation_rate.roll(t) * dt + orientation.roll(t-dt)

orientation_rate.yaw(t) = orientation_acceleration.yaw(t) * dt + orientation_rate.yaw(t-dt)
orientation_rate.pitch(t) = orientation_acceleration.pitch(t) * dt + orientation_rate.pitch(t-dt)
orientation_rate.roll(t) = orientation_acceleration(t) * dt + orientation_rate.roll(t-dt)

Based on the scalar value calculation, the rotation matrices can be freshly set up  on every time step.

**Comment by TBreitenbergerConti** [comment]
Hello,

yes, I agree. We can avoid that drift.
I will give the chance to reply on this or make a different opinion on this topic public by waiting for several days. After this I will add the given scalar equations as example in the commented area.


Von: Ludwig Friedmann <notifications@github.com>
Gesendet: Donnerstag, 13. Dezember 2018 09:19
An: OpenSimulationInterface/open-simulation-interface <open-simulation-interface@noreply.github.com>
Cc: Breitenberger, Thomas <Thomas.Breitenberger@continental-corporation.com>; Mention <mention@noreply.github.com>
Betreff: Re: [OpenSimulationInterface/open-simulation-interface] Added orientation acceleration to message BaseMoving (solves #279) (#291)


Hi @TBreitenbergerConti<https://github.com/TBreitenbergerConti> and @CarloVanDriestenBMW<https://github.com/CarloVanDriestenBMW>,

as matrix multiplication on every timestep results in unnecessary numerical drift and as we're neither interested in relative kinematics nor in time derivatives of rotational matrices - why don't we use examples for the scalar values (which I would recommmend to update)?

For discrete time steps and no higher order integration schemes, the linear scalar formulas would be:

orientation.yaw(t) = orientation_rate.yaw(t) * dt + orientation.yaw(t-dt)
orientation.pitch(t) = orientation_rate.pitch(t) * dt + orientation.pitch(t-dt)
orientation.roll(t) = orientation_rate.roll(t) * dt + orientation.roll(t-dt)

orientation_rate.yaw(t) = orientation_acceleration.yaw(t) * dt + orientation_rate.yaw(t-dt)
orientation_rate.pitch(t) = orientation_acceleration.pitch(t) * dt + orientation_rate.pitch(t-dt)
orientation_rate.roll(t) = orientation_acceleration(t) * dt + orientation_rate.roll(t-dt)

Based on the scalar value calculation, the rotation matrices can be freshly set up on every time step.

**Comment by TBreitenbergerConti** [comment]
Hello again.
With a small correction for orientation roll rate:
orientation_rate.roll(t) = orientation_acceleration.roll(t) * dt + orientation_rate.roll(t-dt)



Von: Ludwig Friedmann <notifications@github.com>
Gesendet: Donnerstag, 13. Dezember 2018 09:19
An: OpenSimulationInterface/open-simulation-interface <open-simulation-interface@noreply.github.com>
Cc: Breitenberger, Thomas <Thomas.Breitenberger@continental-corporation.com>; Mention <mention@noreply.github.com>
Betreff: Re: [OpenSimulationInterface/open-simulation-interface] Added orientation acceleration to message BaseMoving (solves #279) (#291)


Hi @TBreitenbergerConti<https://github.com/TBreitenbergerConti> and @CarloVanDriestenBMW<https://github.com/CarloVanDriestenBMW>,

as matrix multiplication on every timestep results in unnecessary numerical drift and as we're neither interested in relative kinematics nor in time derivatives of rotational matrices - why don't we use examples for the scalar values (which I would recommmend to update)?

For discrete time steps and no higher order integration schemes, the linear scalar formulas would be:

orientation.yaw(t) = orientation_rate.yaw(t) * dt + orientation.yaw(t-dt)
orientation.pitch(t) = orientation_rate.pitch(t) * dt + orientation.pitch(t-dt)
orientation.roll(t) = orientation_rate.roll(t) * dt + orientation.roll(t-dt)

orientation_rate.yaw(t) = orientation_acceleration.yaw(t) * dt + orientation_rate.yaw(t-dt)
orientation_rate.pitch(t) = orientation_acceleration.pitch(t) * dt + orientation_rate.pitch(t-dt)
orientation_rate.roll(t) = orientation_acceleration(t) * dt + orientation_rate.roll(t-dt)

Based on the scalar value calculation, the rotation matrices can be freshly set up on every time step.

**Comment by CarloVanDriestenBMW** [comment]
@TBreitenbergerConti could you update until Thursday evening, please? I would like to release on Friday! :)

**Comment by CarloVanDriestenBMW** [comment]
@carsten-kuebler could I lend your expertise for a moment and ask how to display the formulas in doxygen the best way? As note?

**Pull request merged**